{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Critical manuscript reading\n",
    "\n",
    "You want to work with a proteomics data set. \n",
    "\n",
    "_But which information and which files do you need so you can actually do some analysis?_\n",
    "\n",
    "__Your problem__  \n",
    "Independently whether you <ins>want to re-analyze public data</ins> or <ins>get data from a core facility</ins>, you need a minimum of information do further process them, and _much more information_ to fully understand the data.\n",
    "\n",
    "_So what information is necessary?_\n",
    "\n",
    "This means:\n",
    "- files at different stages (raw spectra, identifications, quantified peptides and proteins)\n",
    "- detailed annotation of the files to relate them with the experimental design\n",
    "- detailed description of the data analysis to reproduce the results and check whether they were processed correctly\n",
    "\n",
    "In order to gather a feeling of MS data _and_ how it usually is described, you will work in groups to assess data availability, annotation and description in 3 publications. One of them will also be used during the next days.\n",
    "\n",
    "Our European EuBIC initiative recently devised a set of guidelines of how to assess proteomics data in publications: https://eubic.github.io/ReproducibleMSGuidelines/\n",
    "\n",
    "📓 We want you now to look into the following papers and take notes which of the guidelines were fulfilled and which not. \n",
    "\n",
    "_Paper 1_: https://www.ncbi.nlm.nih.gov/pubmed/29208753\n",
    "\n",
    "_Paper 2_: https://www.ncbi.nlm.nih.gov/pubmed/30962452\n",
    "\n",
    "_Paper 3_: https://www.ncbi.nlm.nih.gov/pubmed/29621647\n",
    "\n",
    "\n",
    "👨‍💻 Please take one of these papers and add your comments in the cell below.\n",
    "\n",
    "💬 We will discuss the papers afterwards"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Guidelines and space for answers\n",
    "📓 Please add your comments and impressions in the fields below\n",
    "\n",
    "\n",
    "### Conventions Used in the guidelines\n",
    "\n",
    "All items are grouped into three categories:\n",
    "\n",
    "**Bronze 🥉**: Minimal requirements to ensure reproducibility. This constitutes the minimal level of compliance with this guideline.\n",
    "\n",
    "**Silver 🥈**: Intermediate level\n",
    "\n",
    "**Gold 🥇**: Fully documented consideration of all bioinformatics aspects."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Public Data Deposition\n",
    "\n",
    "\n",
    "We consider public data deposition as a vital requirement for any published study. Most of the repositories take part of the ProteomeXChange Consortium.\n",
    "\n",
    "👨‍💻 Which are large data repositories for proteomics data?  \n",
    "👨‍💻 Check out some of the data set in PRIDE.  \n",
    "❔ How well do you think they are described?  \n",
    "\n",
    "\n",
    "👨‍💻 Now carefully read the guidelines and ask the instructors or your peers about the details when you don't understand something  \n",
    "👨‍💻 Look into the assigned publication and its material and give feedback on each guideline\n",
    "\n",
    "<span style=\"background-color:#ffdddd;border-left:6px;\">Note: The guidelines are far from perfect. 😉 Any idea how to improve them is highly appreciated.</span>\n",
    "\n",
    "\n",
    "🏷️ Name: Raw data is deposited in a public repository  \n",
    "Category: Bronze 🥉  \n",
    "🗒️ Description:  The complete RAW data (ie. the mass spectrometer's raw files, such as `RAW` files for Thermo Fisher instruments) of any MS-based study must be deposited in a public repository to ensure reproducibility.\n",
    "Fields: all  \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: The data repository is part of [ProteomeXchange](http://www.proteomexchange.org/)  \n",
    "Category: Bronze 🥉  \n",
    "🗒️ Description: The ProteomeXchange Consortium coordinates standard data submission and dissemination guidelines of proteomics data. It encourages open data policies in the field of proteomics and guarantees the long-term maintainability of the data.\n",
    "Fields:  Proteomics, ProteoGenomics  \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: Data should be deposited in one of the following repositories: Metabolights, MassIVE  \n",
    "Category: Silver 🥈   \n",
    "🗒️ Description: [Metabolights](https://www.ebi.ac.uk/metabolights/) and [MassIVE](https://massive.ucsd.edu) are the two major archives for metabolomics data in the world. These two archives implement standard protocols for data submission and dissemination of metabolomics data.      \n",
    "Fields: Metabolomics  \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: Experimental design is deposited in a public repository  \n",
    "Category: Bronze 🥉  \n",
    "🗒️ Description:  An accompanying file is made available which describes the experimental design in terms of sample id per raw file / isobaric channel, and fraction identifier, as well as any additional factors relevant to the experiment. This file should be deposited alongside the data in the public repository or must alternatively be submitted as a supplementary file with the manuscript.This file should indicate factors potentially relevant to batch effects (f.e. machine id, lab id, run date and time). If supported, this information should be made available using the repository’s standard method.\n",
    "Fields: all    \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: The identifier of the data submission is mentioned in the manuscript  \n",
    "Category: Bronze 🥉  \n",
    "🗒️ Description: \n",
    "Fields: all     \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: The data must be available at the time of submission for review and must be available publicly at the time of publication.\n",
    "Category: Bronze 🥉  \n",
    "🗒️ Description: The assessment of raw data, or at least the way the data was submitted, is vital to the review process of a scientific publication. This\n",
    "allows reviewers to ensure that the data was deposited in a re-usable way.\n",
    "Fields: all    \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: Linked to matched data, e.g. transcriptomics is provided  \n",
    "Category: Silver 🥈  \n",
    "🗒️ Description: The accession of the complementary omics data should be provided in the main manuscript. These data should be \n",
    "available in another public repository (e.g. RNA-seq dataset in ArrayExpress). \n",
    "Example: [PXD008960](https://www.ebi.ac.uk/pride/archive/projects/PXD008960) by Ren *et al.* is an example of a proteo-genomics study where RNA-seq data\n",
    "is linked to proteomics data.\n",
    "Fields: all   \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: The processed data is deposited in a public repository  \n",
    "Category: Silver 🥈    \n",
    "🗒️ Description: The **processed** result files (ie. search result files indicating identified PSMs, peptides, and proteins) are deposited in the same public repository as the raw data.\n",
    "Fields: all   \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: For meta-analysis papers, the identifiers of all reanalyzed datasets are mentioned in the manuscript    \n",
    "Category: Bronze 🥉    \n",
    "🗒️ Description: For meta-analysis manuscripts where multiple public datasets are reanalyzed with a different approach or bioinformatics workflow, the identifier and the publication of each of these datasets should be cited in the manuscript.    \n",
    "Fields: all    \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: Acquisition method files are made available together with the Raw data     \n",
    "Category: Silver 🥈  \n",
    "🗒️ Description:    \n",
    "Fields: Proteomics, ProteoGenomics  \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: The mass spectra are provided in the HUPO-PSI standard file format mzML  \n",
    "Category: Gold 🥇   \n",
    "🗒️ Description: [mzML](http://www.psidev.info/mzML) is a HUPO-PSI data format that store mass spectrometry data (e.g mass spectra, chromatograms) using controlled vocabulary terms. mzML, is a well tested open-source and XML-based format.   \n",
    "Fields: Proteomics, ProteoGenomics, Metabolomics, Lipidomics\n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: The identification information (PSMs / Petides / Proteins) are provided in the HUPO-PSI format mzIdentML or mzTab  \n",
    "Category: Gold 🥇  \n",
    "🗒️ Description: This relates to the search results. As most proteomics search engines support [mzIdentML](http://www.psidev.info/mzidentml) and [mzTab](http://www.psidev.info/mztab), the files generated\n",
    "by the search engine should be uploaded to the same repository as the mass spectrometer's raw data.\n",
    "Fields: Proteomics, ProteoGenomics  \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: The quantitative information is made available in the HUPO-PSI standard file format mzTab   \n",
    "Category: Gold 🥇  \n",
    "🗒️ Description: [mzTab](http://www.psidev.info/mztab) is a HUPO-PSI standard file fomat that store proteomics and metabolomics identification and quantification results including: Peptide spectrum matches, Peptides, metabolites and proteins. \n",
    "Fields: Proteomics, ProteoGenomics, Metabolomics  \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: Samples are annotated with metadata within the HUPO-PSI files (mzTab) or other standards for experimental design such as IsaTab  \n",
    "Category: Gold 🥇  \n",
    "🗒️ Description:  \n",
    "Fields: Proteomics, ProteoGenomics, Metabolomics, Lipidomics\n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: The identifier to the genomics (e.g. GWAS or RNA-seq) dataset is cited in the manuscript  \n",
    "Category: Bronze 🥉  \n",
    "🗒️ Description: In ProteoGenomics studies, sequencing data is an integral part of the dataset. Therefore, this data should also be deposited\n",
    "to a public repository and the respective dataset cited in the manuscript.\n",
    "Fields: ProteoGenomics  \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: All 'omics datasets are linked at the repository level\n",
    "Category: Silver 🥈\n",
    "🗒️ Description: For example PRIDE (for proteomics data) and ArrayExpress (for expression data) support the linking of RNA-sequencing and\n",
    "proteomics datasets. If this feature is supported by the chosen repository, all 'omics datasets should also be linked at the repository\n",
    "level.\n",
    "Fields: ProteoGenomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Basic Bioinformatics Analysis Description\n",
    "\n",
    "This section refers to the requirements of how the bioinformatic analysis should be described in a manuscript.\n",
    "\n",
    "👨‍💻 Carefully read the guidelines and ask the instructors or your peers about the details when you don't understand something  \n",
    "👨‍💻 Look into the assigned publication and its material and give feedback on each guideline\n",
    "\n",
    "<span style=\"background-color:#ffdddd;border-left:6px;\">Note: The guidelines are far from perfect. 😉 Any idea how to improve them is highly appreciated.</span>\n",
    "\n",
    "\n",
    "🏷️ Name: All used software is named including the exact version and the availability\n",
    "Category: Bronze 🥉    \n",
    "🗒️ Description: The used software must be unambiguously named. This includes, for example, the used download link for public software (including the citation of the corresponding publication if applicable) or reference to the company.\n",
    "The provided information should be sufficient for anyone to retrieve the exact same piece of software (if still available). \n",
    "Fields: all   \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: Any non-publicly available / custom scripts are available to the reviewers      \n",
    "Category: Bronze 🥉      \n",
    "🗒️ Description: The term `scripts` refers to any short piece of code that was created for the study. Most commonly, these will be custom, in-house\n",
    "R, Python, MatLab, or Perl scripts. This explicitly does not mean that commercial software must be available free-of-charge to the reviewers.\n",
    "Fields: all    \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: Newly developed or closed-source in-house developed software is made available to the reviewers for testing\n",
    "Category: Bronze 🥉     \n",
    "🗒️ Description: This refers to software that is developed by the authors / research group / institute.\n",
    "Fields: all    \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: If the software is made publicly available, downloads for specific versions of the software are provided, which include the exact version used in the manuscript\n",
    "Category: Bronze 🥉     \n",
    "🗒️ Description: The way to distribute the software must support to download specific versions of the software to enable researchers to reproduce the exact same results.\n",
    "Fields: all    \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: All external resources (UniProt, GO, Ensembl, Reactome, etc.) are referenced and the version/release date/date of access documented    \n",
    "Category: Bronze 🥉    \n",
    "🗒️ Description: During the downstream analysis different databases are commonly use, for example Uniprot, Intact, Reactome. Some of these databases are used througth R packages, and desktop applications. However, the version of the tool is not enough to reproduce the original results, because the same tool version can be use with different dtabase versions. We recommended to include the version/release/date of the data provider in addition to the tool version.    \n",
    "Fields: all   \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: The operating system and its version under which the analysis was carried out is specified    \n",
    "Category: Bronze 🥉     \n",
    "🗒️ Description: As shown by [DiTommaso et al](https://www.nature.com/articles/nbt.3820) the choice of operating system may have a considerable effect on the generated results even if the same software was used.\n",
    "Fields: all    \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: Newly developed software/custom scripts are made publicly available and deposited to a public code hosting platform\n",
    "Category: Silver 🥈    \n",
    "🗒️ Description: All software, including custom scripts (see above) should ideally be deposited at a public hosting platform, such as\n",
    " GitHub, Bitbucket, or GitLab. This should ensure the long-term availability of the software as well as the option to enable researchers\n",
    " to retrieve specific versions of the software in the future.\n",
    "Fields: all   \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: A README file is provided that specifies the system requirements, installation guide, and instructions for use\n",
    "Category: Silver 🥈  \n",
    "🗒️ Description: This refers to any custom / new software developed by the authors.\n",
    "Fields: all \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: The used software version can be identified through a DOI\n",
    "Category: Gold 🥇  \n",
    "🗒️ Description: Services like https://zenodo.org provide DOIs for software. This simplifies the citation and retrieval of specific software versions.\n",
    "Fields: all    \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: Demo data is provided for newly developed software\n",
    "Category: Gold 🥇  \n",
    "🗒️ Description: Next to a README file (see above), a demo dataset is distributed with the software together with a description how to run and what to expect from the provided data. This resembles the idea of vignettes in [R Bioconductor](https://www.bioconductor.org/) packages. If applicable, [Jupyter notebooks](https://jupyter.org) provide a very good platform to create and distribute such example analyses.\n",
    "Fields: all  \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Workflow software and containers  \n",
    "\n",
    "👨‍💻 Carefully read the guidelines and ask the instructors or your peers about the details when you don't understand something  \n",
    "👨‍💻 Look into the assigned publication and its material and give feedback on each guideline\n",
    "\n",
    "<span style=\"background-color:#ffdddd;border-left:6px;\">Note: The guidelines are far from perfect. 😉 Any idea how to improve them is highly appreciated.</span>\n",
    "\n",
    "#### Workflow software\n",
    "This section refers to cases where a workflow software, such as ProteomeDiscoverer, Galaxy, or OpenMS was used to perform the analysis.\n",
    "\n",
    "🏷️ Name: The workflow file is made available at the time of review  \n",
    "Category: Bronze 🥉  \n",
    "Description:\n",
    "Fields: all   \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: Workflow files are deposited in the same repository as the public data or the software or submitted as supplementary data   \n",
    "Category: Silver 🥈  \n",
    "🗒️ Description: In case of newly developed software that integrates into a workflow system, such a workflow file can be distributed  as a test suite for the\n",
    "software package.\n",
    "Example: The ProteomeDiscoverer nodes developed by the Protein Chemistry Facility at the IMP provide example workflows together with their respective nodes.\n",
    "See the [MSAmanda node](http://ms.imp.ac.at/index.php?action=ms-amanda) as an example ([Dorfer et al. J Proteome Res. 2014, 13(8):3679-84](http://pubs.acs.org/articlesonrequest/AOR-6DyVQ3j4YTcGXyaskJvi))\n",
    "Fields: all   \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "### Containers\n",
    "\n",
    "Special requirements when using containerized software (such as Docker containers).\n",
    "\n",
    "❔ Do you know what a Docker container is? You can read more about them [here](https://stackabuse.com/docker-a-high-level-introduction/)\n",
    "\n",
    "🏷️ Name: If containers are used in the analysis, they should be referenced following, for example, the BioContainers guidelines\n",
    "Category: Bronze 🥉\n",
    "🗒️ Description: If containers, such as Docker or Singularity containers, are used in the analysis\n",
    "they should be referenced through stable version numbers. This explicitly relates to not\n",
    "using the \":latest\" tag for Docker images as these are bound to change upon new releases\n",
    "of the software. Detailed suggestions can be found in the BioContainers documentation\n",
    "Fields: all\n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: If containers are used in the analysis, these should be available in a public repository using non-personal namespaces\n",
    "Category: Silver 🥈\n",
    "🗒️ Description: If containers, such as Docker or Singularity containers, are used in the analysis\n",
    "they should be available through a public repository, such as Docker Hub. The namespace used to make this image publicly available should not be under a \"private\", user-based namespace but should use some kind of institutional namespace where long-term availability is ensured. This addresses the risk, that if private namespaces are used and the person changes careers, the namespace might be deleted and the images thus lost.\n",
    "Fields: all\n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: Containers should be available in dedicated repositories such as BioContainers\n",
    "Category: Gold 🥇\n",
    "🗒️ Description: Dedicated namespaces for bioinformatics tools ensure minimum standards of the containers and their long-term availability. Additionally, they have mechanisms in place to also support a wider range of platforms, such as BioConda.\n",
    "Fields: all\n",
    "\n",
    "<ins>Please add your evaluation here:</ins>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### From raw data to quantitative values\n",
    "\n",
    "👨‍💻 Carefully read the guidelines and ask the instructors or your peers about the details when you don't understand something  \n",
    "👨‍💻 Look into the assigned publication and its material and give feedback on each guideline\n",
    "\n",
    "<span style=\"background-color:#ffdddd;border-left:6px;\">Note: The guidelines are far from perfect. 😉 Any idea how to improve them is highly appreciated.</span>\n",
    "\n",
    "#### RAW File Processing\n",
    "In proteomics, raw file processing refers to all steps before generating peak lists for the identification step.\n",
    "\n",
    "❔ Do you know what peak list is? Ask us!\n",
    "\n",
    "🏷️ Name: If RAW file processing was performed by a separate tool, this tool including all settings are described   \n",
    "Category: Bronze 🥉    \n",
    "🗒️ Description: This should include all settings for signal detection (such as MS and MS/MS peak picking, deisotoping, feature detection) settings if available    \n",
    "Fields: all    \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: If the software used is not publicly available, the processed RAW data is deposited alongside the original RAW data\n",
    "Category: Bronze 🥉  \n",
    "🗒️ Description: This is only required if custom software was used to process the RAW data.\n",
    "Fields: all \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "####  Manual Curation\n",
    "\n",
    "🏷️ Name: If the data were manually curated (for example using Skyline) the curated dataset is deposited alongside the RAW data in a public repository    \n",
    "Category: Bronze 🥉    \n",
    "🗒️ Description: Manual curation refers to any manipulation of the original RAW data that was not performed using an automated process. Therefore, the\n",
    "process cannot be sufficiently described and reproduced based on used software settings alone. In such a case, any intermediate data created from such\n",
    "a curation step must be made available to ensure reproducibility.\n",
    "Fields: all\n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: Un-curated data is made available alongside the final one    \n",
    "Category: Gold 🥇    \n",
    "🗒️ Description: If different data than the original RAW files were used as starting point for the manual curation step, this data should also be made\n",
    "publicly available.\n",
    "Fields: all   \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "#### Search MS-based Proteomics\n",
    "\n",
    "Protein identification is based on database searches in most cases. The used software (search engines) and their parameter settings can lead to quite different results. Therefore, we need to have this process well described.\n",
    "\n",
    "🏷️ Name: The protein sequence database used including the exact version/date of download (if a release version is not available) are described  \n",
    "Category: Bronze 🥉    \n",
    "🗒️ Description: During the protein identification step, the spectrum identification search engine algorithm commonly uses a  protein sequence FASTA database from public resources (e.g. ENSEMBL, Uniprot). If the original FASTA file from the resource is not transform/change and the release number of the file is not available; the exact version/date of the downloaded FASTA file should be provided.  \n",
    "Fields: Proteomics, ProteoGenomics  \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: The method used to create a decoy database is specified  \n",
    "Category: Bronze 🥉   \n",
    "🗒️ Description: During the peptide identification step, the search engine uses a decoy database to statistically access the quality of the peptide identifications. Multiple methods are available to generate the decoy databases (e.g reverse, random, etc). We recommend to mention the strategy used to generate the decoy database in the main manuscript.    \n",
    "Fields: Proteomics, ProteoGenomics  \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: If a custom (ie. non-publicly available) database was used, this database is deposited in the public repository or submitted as supplementary data  \n",
    "Category: Bronze 🥉  \n",
    "🗒️ Description:  This includes databases, where decoy databases were generated using randomization. In ProteoGenomics studies sequencing data is often used to create custom protein databases (FASTA files) for the subsequent identification step. In these cases, such custom FASTA files should be made available to ensure reproducibility. In case of human studies, the files can be access-protected.\n",
    "Fields:  Proteomics, ProteoGenomics \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: Custom spectral libraries should be publicly available\n",
    "Category: Bronze 🥉   \n",
    "🗒️ Description: If custom spectral libraries are used for peptide / protein identification, the libraries should be deposited in the public repository. If the libraries are publicly available, the source, version (if available) or date of download should be mentioned in the manuscript.\n",
    "Fields: Proteomics, ProteoGenomics  \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: The search engine(s) including their exact versions is/are specified  \n",
    "Category: Bronze 🥉  \n",
    "🗒️ Description: During the identification step a `search engine` is used to map the obtained mass spectra to the corresponding peptide sequence. In this rule, we recommend to report the software version in the main manuscript.   \n",
    "Fields: Proteomics, ProteoGenomics\n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: All relevant, non-default settings, such as precursor ion tolerance, fragment ion tolerance, in-silico digestion rule (ie. enzyme), allowed missed cleavages, static and variable post-translational modifications are specified  \n",
    "Category: Bronze 🥉  \n",
    "🗒️ Description:  \n",
    "Fields: Proteomics, ProteoGenomics  \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: The methods used to determine the identification false discovery rates (at PSM, peptide and/or protein levels) and the respective thresholds \n",
    "are specified   \n",
    "Category: Bronze 🥉   \n",
    "🗒️ Description:    \n",
    "Fields: Proteomics, ProteoGenomics    \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: When available, the used parameter/configuration file is deposited alongside the data in a public repository or alternatively submitted as supplementary data   \n",
    "Category: Silver 🥈   \n",
    "🗒️ Description: Several search engines are configured through parameter files. If this is supported by the used software, the used parameter files should\n",
    "be either deposited in a public repository (alongside the RAW data) or submitted as supplementary data.\n",
    "Fields: Proteomics, ProteoGenomics   \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "#### MS-based Quantitative\n",
    "\n",
    "This section refers to the step in the analysis where quantitative values are derived for the measured molecules. In proteomics,\n",
    "for example, the result of this process is a table with protein expressions and their abundance in the various samples.\n",
    "\n",
    "🏷️ Name: The software used to extract quantitative values including its parameters and version is specified    \n",
    "Category: Bronze 🥉   \n",
    "🗒️ Description: This refers to the software to create abundance values from the RAW spectra. For example, the software used to\n",
    "extract precursor ion intensities, or reporter ion intensities etc.\n",
    "Fields: all\n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: The method used to aggregate PSM level quantitative values on the peptide and protein levels is specified   \n",
    "Category: Bronze 🥉   \n",
    "🗒️ Description: This refers to the step where the measured PSM-level quantitative values are aggregated at the protein level.\n",
    "Fields: Proteomics, ProteoGenomics   \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: The method used to perform normalisation is specified including at which level this normalisation was performed (PSM, peptide, or protein)    \n",
    "Category: Bronze 🥉   \n",
    "Description:\n",
    "Fields: Proteomics, ProteoGenomics    \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: The method used to filter and/or impute missing values is specified including at which level this filtering/imputation was performed (PSM, peptide, or protein)    \n",
    "Category: Bronze 🥉   \n",
    "🗒️ Description:   \n",
    "Fields: Proteomics, ProteoGenomics   \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Statistical Analysis\n",
    "\n",
    "This section refers to the step where the obtained quantitative data is analysed to, generally, identify differentially\n",
    "expressed molecules between the different groups of samples.\n",
    "\n",
    "👨‍💻 Carefully read the guidelines and ask the instructors or your peers about the details when you don't understand something  \n",
    "👨‍💻 Look into the assigned publication and its material and give feedback on each guideline\n",
    "\n",
    "<span style=\"background-color:#ffdddd;border-left:6px;\">Note: The guidelines are far from perfect. 😉 Any idea how to improve them is highly appreciated.</span>\n",
    "\n",
    "\n",
    "🏷️ Name: The software used to perform the statistical analysis (ie. differential expression analysis) is specified including its version    \n",
    "Category: Bronze 🥉   \n",
    "🗒️ Description:   \n",
    "Fields: all\n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: The method used to determine the sample size and the power of the study is specified (if applicable)   \n",
    "Category: Bronze 🥉   \n",
    "🗒️ Description:   \n",
    "Fields: all\n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: Any statistical test is named and the resulting p-value/threshold specified    \n",
    "Category: Bronze 🥉   \n",
    "🗒️ Description: This includes the information whether the test is one- or two-sided, paired or unpaired, as well as all additional parameters passed to the test.\n",
    "Fields: all\n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: The method used to correct for multiple testing is specified   \n",
    "Category: Bronze 🥉   \n",
    "🗒️ Description: If correction for multiple testing was not performed, this needs to be explicitly justified. \n",
    "Fields: all\n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: If the software supports loading external libraries, all used libraries including their versions are specified or provided as supplementary file\n",
    "Category: Bronze 🥉   \n",
    "🗒️ Description: This refers to libraries commonly used in R, Python, or MatLab. In R, the complete environment can be easily exported using the\n",
    "`sessionInfo` command.\n",
    "Fields: all\n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: If any filtering was applied / measurement(s) excluded either manually or as part of the software algorithms, this must be clearly stated    \n",
    "Category: Bronze 🥉   \n",
    "🗒️ Description:    \n",
    "Fields: all\n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: Any transformation of data (e.g., log transformation) manually or as part of a software function/test is specified including the order that these transformations were performed   \n",
    "Category: Bronze 🥉  \n",
    "🗒️ Description:   \n",
    "Fields: Proteomics, ProteoGenomics   \n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: If the software supports the export of the statistical analysis steps (for example R scripts) these are made available for review  \n",
    "Category: Silver 🥈   \n",
    "🗒️ Description: We recommend the use of [Jupyter notebooks](https://jupyter.org) to record and share a complete analysis workflow.\n",
    "Fields: all\n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: If a model with multiple parameters is created, the formula for the said model is declared  \n",
    "Category: Silver 🥈  \n",
    "🗒️ Description: This most commonly refers to the use of linear models, used, for example, in [limma](https://www.bioconductor.org/packages/release/bioc/html/limma.html) for the analysis of differentially expressed proteins.\n",
    "Fields: all\n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: If the software supports the export of the statistical analysis steps (for example R scripts) these are submitted as supplementary data  \n",
    "Category: Gold 🥇 \n",
    "🗒️ Description:  We recommend the use of [Jupyter notebooks](https://jupyter.org) to record share a complete analysis workflow.\n",
    "Fields: all\n",
    "\n",
    "<ins>Please add your evaluation here:</ins>\n",
    "\n",
    "🏷️ Name: Underlying data for all figures (where applicable) is made available  \n",
    "Category: Gold 🥇  \n",
    "🗒️ Description: During the downstream analysis multiple figures complement the final results of the manuscript. We recommend to make the underlying data of all figures available (where applicable) as supplementary information of the manuscript or as part of the dataset. Ideally, the code to create these figures should be made available as well (for example using [Jupyter notebooks](https://jupyter.org)).\n",
    "Fields: all\n",
    "\n",
    "<ins>Please add your evaluation here:</ins>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
