---
layout: page
title: Proteogeonomics
schemadotorg:
  "@context": http://schema.org/
  "@type": CreativeWork
  "genre": TrainingMaterial
  isPartOf:
      url: "https://gtpb.github.io/IBIP19/"
      name: "IBIP19 - Integrative Biological Interpretation using Proteomics"
---

Proteogenomics is the scientific field at the interface between proteomics and genomics [(1)](#references). First, proteogenomic applications focused on improving genomic annotation through the use of proteomics data [(2)](#references). Over the years, proteogenomics rapidly developed as a rich scientific field for the joint analysis of genomic and proteomic information, generating fundamental knowledge on the interplay between genome and proteome.

Due to the novelty of the field, the complexity of the analyses, and the volume of data involved, proteogenomics poses multiple bioinformatic challenges [(3)](#references). These tutorials present methods that can be used to analyse results from proteogenomic pipelines.

Introduction slides are available [here](proteogenomics/resources/slides/day2.pptx)

## A word of caution

Please note that this tutorial starts from the results of proteomic and genomic analyses, and does not cover the details of respective bioinformatic pipelines, like the calling of variants. Please note also that the generation and analysis of proteomic data for proteogenomic application differ from standard proteomic analyses, requiring specific methods and expertise. Notably, the identification of non-canonical genetic products of transcription/translation and the calling of variants in proteomics data require high coverage and high resolution proteomic analyses. These analyses also require the use of complex bioinformatic workflows in high performance environments. Finally, they can benefit from tailoring the experimental workflow in relationship to the bioinformatic analysis [(4)](#references).

Note that establishing a proteogenomic workflow, in both wet- and dry-lab is a long-term endeavor. If you plan to conduct a proteogenomic experiment, please give particular attention to the following questions:
- Do I have enough sample material, replicates, and controls?
- Has the proteomics lab sufficient instrumentation and expertise to run the samples?
- Is the proteogenomic bioinformatic pipeline in place and able to handle the volume of data generated by my experiment?

When in doubt, consider running a small pilot experiment to benchmark the performance of your analytical set-up. Also, prioritize proteomics labs that have experience with proteogenomic analyses and analyzing the data.

The present tutorials focus on human samples and resources. While the methods presented are generic, differences are to be expected when working with other organisms, notably on the level of genomic and proteomic data annotation. Please note that working with multiple organisms, _e.g._ in metaproteogenomic analyses [(5)](#references), is yet another level of complexity, posing multiple challenges for the interpretation of the data [(6)](#references).

## Proteogenomics Lexicon

Analysing proteogenomics data requires familiarity with both genomic and proteomic fields, which can leave scientists discombobulated. We provide here a short lexicon of the terms and abbreviations the reader will encounter in the literature. Please do not hesitate to suggest terms that should be added in our [issue tracker](https://github.com/GTPB/IBIP19/issues).

![From_Genes_To_Proteoforms](proteogenomics/resources/images/Insulin.png?raw=true "Insulin from gene to proteoforms")

> The production of human insulin from the transcript INS-201 according to Ensembl release 97. (a) The transcript is coded by three exons on chromosome eleven, colored in yellow, orange, and dark red. The translated sequence is underlined. Sequence variation in the translated sequence can result in sequence variations in the amino acid sequence, and hence different proteoforms. (b) The sequence obtained after translation represents the raw proteoform of insulin, called preproinsulin, which requires posttranslational maturation to obtain the mature form of insulin. Amino acids are colored according to the coding exons and the residue overlapping splice site is underlined. (c) The signal peptide is cleaved, yielding proinsulin, and cysteines cross-linked by disulphide bonds, making a new proteoform of insulin. (d) Proteases cleave a large fragment of the sequence, the C-peptide. The C-peptide is often used as a proxy to measure insulin production. (e) Proteases cleave pairs of amino-acids, yielding the mature form of insulin. (f) The mature form of insulin consists of two cross-linked peptides. It can be further modified, yielding even more proteoforms. From [(7)](#references), adapted from [en.wikipedia.org/wiki/Insulin](https://en.wikipedia.org/wiki/Insulin#/media/File:Insulin_path.svg).

- _Variant, Mutation, Alteration_: A genetic variant or mutation refers to a variation in the genetic sequence. When the variation is not inherited, it is called alteration. Variants involving the substitution of a single nucleotide are called single-nucleotide polymorphism (SNP) or single-nucleotide variant (SNV). SNP refers to variants where each version is carried by more than 1% of the population. When a variation of one amino acid is detected in the proteome, it is referred to as single amino acid variant (SAV or SAAV). Variants involving the deletion or insertion of genetic code are called indels. Variants involving the repeat of sections of the genome are examples of structural variation, and called Copy-number variation (CNV). When the variation is not inherited, it is called copy-number alteration (CNA).

- _Peptides, Proteins, isoforms, and proteoforms_: peptides and proteins are short and long chains of amino acids, respectively. Proteins are generally associated to a gene which codes their amino acid chain. Differences in splicing yield different isoforms for most proteins. During their lifetime, proteins undergo structural modification: cleavage, folding, cross-linking, post-translational modification (PTM), etc. These modifications yield very different forms for each protein, called proteoforms [(8)](#references).

- _Non-canonical genetic product_: Peptide or protein produced from regions of the genome that are canonically non-coding.

## Legend

You will find the following icons throughout the text:

* :pencil2: : You need to do something
* :speech_balloon: : Something you might want to discuss with someone else (or yourself).
* :thought_balloon: : A question to trigger your attention on an important point of detail. Clicking the icon takes you to the answer and more literature on the subject. Clicking the icon in the answer takes you back to the question.

Please note that questions rarely trigger a yes/no answer. The answers provided represent important elements to take into account when analyzing data, but are by no means exhaustive or universal.


## Processing

This tutorial is organized in notebooks that contains [R](r-project.org) code that can be run directly from the _Rmd_ file. It assumes that the R working directory is the proteogenomics folder of the repository, _e.g._ `/myfolder/IBIP19/pages/proteogenomics`. We recommend using RStudio to run this tutorial.


## Libraries

You will need the following libraries, please make sure that they are installed.

- We will use [tidyr](tidyr.tidyverse.org) to import data, we recommend this [cheat sheet](https://github.com/rstudio/cheatsheets/blob/master/data-import.pdf).
- We will use [dplyr](dplyr.tidyverse.org) to transform data, we recommend this [cheat sheet](https://github.com/rstudio/cheatsheets/blob/master/data-transformation.pdf).
- We will use [ggplot2](ggplot2.tidyverse.org) to plot data, we recommend this [cheat sheet](https://github.com/rstudio/cheatsheets/blob/master/data-visualization-2.1.pdf).
- We will use [gtable](github.com/r-lib/gtable) to organize plots.
- We will use [gamlss](https://www.gamlss.com/) for normalization.
- We will use [mclust](https://mclust-org.github.io) for Gaussian mixture modeling.
- We will use [igraph](https://igraph.org) for graph manipulation.
- We will use [conflicted](https://github.com/r-lib/conflicted) to manage conflict resolution to function names between packages.

*Warning*: [conflicted](https://github.com/r-lib/conflicted) is not available in cran yet, you will need to install it using [devtools](https://devtools.r-lib.org/). See [the installation instructions](https://github.com/r-lib/conflicted#installation) for more details.

## Tutorials

- [1. Novel peptides](proteogenomics/Novel Peptides.ipynb): mapping proteomics results to non-coding regions of the genome.

- [2. Variation Analysis](proteogenomics/Variation Analysis.ipynb): studying sequence and splicing variation in proteomics data.

- [3. CNA-Protein](proteogenomics/CNA-Protein.ipynb): linking structural variants and protein levels to study CNA dampening/silencing.

- [4. RNA-Protein](proteogenomics/RNA-Protein.ipynb): comparing RNA and protein levels to identify key biological mechanisms.


## References

(1) [_Proteogenomics: concepts, applications and computational strategies_](https://www.ncbi.nlm.nih.gov/pubmed/25357241)

(2) [_Proteogenomic mapping as a complementary method to perform genome annotation_](https://www.ncbi.nlm.nih.gov/pubmed/14730672)

(3) [_Proteogenomics from a bioinformatics angle: A growing field_](https://www.ncbi.nlm.nih.gov/pubmed/26670565)

(4) [_HiRIEF LC-MS enables deep proteome coverage and unbiased proteogenomics_](https://www.ncbi.nlm.nih.gov/pubmed/24240322)

(5) [_Bioinformatic progress and applications in metaproteogenomics for bridging the gap between genomic sequences and metabolic functions in microbial communities_](https://www.ncbi.nlm.nih.gov/pubmed/23625762)

(6) [_Challenges and promise at the interface of metaproteomics and genomics: an overview of recent progress in metaproteogenomic data analysis_](https://www.ncbi.nlm.nih.gov/pubmed/31002542)

(7) [_Proteomics_](rcs.org) in [_Processing Metabolomics and Proteomics Data_](rcs.org)

(8) [_Proteoform: a single term describing protein complexity_](https://www.ncbi.nlm.nih.gov/pubmed/23443629)
